tests/test_web_response.py

```diff
@@ -1171,3 +1173,36 @@ def test_text_is_json_encoded(self) -> None:
     def test_content_type_is_overrideable(self) -> None:
         resp = json_response({"foo": 42}, content_type="application/vnd.json+api")
         assert "application/vnd.json+api" == resp.content_type
+
+
+@pytest.mark.skipif(
+    not bool(os.environ.get("PYTHONASYNCIODEBUG")),
+    reason="Only run the test with debug mode",
+)
+async def test_warn_large_cookie(buf: Any, writer: Any) -> None:
+    resp = Response()
+    resp.set_cookie("foo", "8" * 4070, max_age=2600)  # No warning
+    req = make_request("GET", "/", writer=writer)
+
+    await resp.prepare(req)
+    await resp.write_eof()
+
+    cookie = re.search(b"Set-Cookie: (.*?)\r\n", buf).group(1)
+    assert len(cookie) == 4096
+    buf[:] = b""
+
+    resp = Response()
+
+    with pytest.warns(
+        UserWarning,
+        match="The size of is too large, it might get ignored by the client.",
+    ) as record:
+        resp.set_cookie("foo", "8" * 4071, max_age=2600)
```

Comment:
It's probably better to do all parts of the check in the same place rather than in other part of the test
```suggestion
    ) as warnings_caught:
        resp.set_cookie("foo", "8" * 4071, max_age=2600)
    assert len(warnings_caught) == 1

```

Reply:
I don't think there's really any need to check the record anyway. `with pytest.warns(..):` will fail if no warning happens, so there's no real need to check how many warnings were caught.

Reply:
But we still need to check that no warnings happened.

Reply:
This helper fails if there's no warnings.